src/lib.rs

```diff
@@ -4,6 +4,8 @@
 
 use std::error::Error;
 
```

Comment:
Nit: please remove the blank line between all the imports so that clippy can format them properly.

Reply:
:+1: